cmd/soroban-cli/src/commands/tx/mod.rs

```diff
@@ -0,0 +1,48 @@
+use clap::Parser;
+
+pub mod send;
+pub mod sign;
+pub mod simulate;
+pub mod xdr;
+
+use stellar_xdr::cli as xdr_cli;
+
+#[derive(Debug, Parser)]
+pub enum Cmd {
+    /// Add a new identity (keypair, ledger, macOS keychain)
+    Inspect(xdr_cli::Root),
```

Comment:
It's a bit odd that `tx inspect` works like this: `tx inspect decode --type TransactionEnvelope`.

I think we should replace it with `tx decode` and `tx encode` where the type parameter is already filled in.

See this issue that details the commands:
- https://github.com/stellar/soroban-cli/issues/1180

Reply:
`decode::Cmd` is not [public](https://github.com/stellar/rs-stellar-xdr/blob/efe941d4f0dc55d555273ff7da2b46626f7f985b/src/cli/mod.rs#L1) so I went with root, which is as you point out less than optimal.

Reply:
Making it public here:
- https://github.com/stellar/rs-stellar-xdr/pull/369

We could remove the inspect/encode/decode from this PR and revisit it once that is merged.